### DISCUSSION TITLE
Unable to Redirect to Observation Tab Upon Clicking

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#705] [General Discussion]
Hello Team,

We are on Sunbird release 5.0.I am facing an issue where users are unable to redirect to the observation tab. This issue occurs specifically when the profile role is changed to teacher and there is "Browse Other Categories" in the home tab.
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/db287121-09e0-4f5b-875e-58f95c6a6d43)

However, in the staging environment, clicking on the observation tab successfully redirects users.
Has anyone else encountered this issue?
Any suggestions on how to resolve this would be appreciated.

cc: @AmiableAnil @Hari-stackroute @ashokreddy1208 @vinukumar-vs @rajeevsathish @rajeshkumaravel @akulavasu @amitpriyadarshi

**Comment by rajeevsathish** [comment]
@kiranharidas187 can you please have a look on this discussion and help them

**Comment by kiranharidas187** [comment]
@siddhishinde0723  Please provide the response for following curl from your environment.

`curl '<YOUR_BASE_URL>/api/data/v1/form/read' \
  -H 'accept: application/json' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'content-type: application/json' \
  -H 'cookie: connect.sid=s%3AEbvUnRcHeb_JVEL5nOcK9G_2YTfezQ5E.cQ89MpnegHQmSQRmm4T3naNHVK38VkFh5TmxRRjSms4; __z_a=3101761002337506254533750; _ga=GA1.1.571400613.1712551782; _ga_9Z3LDJ42BE=GS1.1.1712551782.1.0.1712551782.0.0.0' \
  -H 'origin: <YOUR_BASE_URL>' \
  -H 'referer: <YOUR_BASE_URL>/explore?board=CBSE%2FNCERT&gradeLevel=Class%201&gradeLevel=Class%202&id=ncert_k-12&selectedTab=home' \
  -H 'sec-ch-ua: "Google Chrome";v="123", "Not:A-Brand";v="8", "Chromium";v="123"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'ts: 2024-04-08T10:19:42+05:30' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' \
  -H 'x-app-id: prod.diksha.portal' \
  -H 'x-app-version: 5.1.0' \
  -H 'x-channel-id: ORG_001' \
  -H 'x-device-id: e5b5922b4dfe25b90638b4eb7ef9603b' \
  -H 'x-msgid: 784f69c4-9c73-a8d3-23b7-e428d7fdf3b5' \
  -H 'x-org-code: ORG_001' \
  -H 'x-request-id: 784f69c4-9c73-a8d3-23b7-e428d7fdf3b5' \
  -H 'x-session-id: e810288a-4a9f-529c-fe24-418090c31732' \
  -H 'x-source: web' \
  --data-raw '{"request":{"type":"category","action":"homeListing","subType":"targetedCategory","rootOrgId":"0126684405014528002"}}'`

**Reply by siddhishinde0723** [reply]
@kiranharidas187  here is the response for the form read.

`{
    "id": "api.form.read",
    "params": {
        "resmsgid": "94833796-24a8-417f-bb7e-347d5b95b6e3",
        "msgid": "f2267f62-3a0e-4997-ad56-fc06d2f0623c",
        "status": "successful"
    },
    "responseCode": "OK",
    "result": {
        "form": {
            "type": "category",
            "subtype": "targetedcategory",
            "action": "homelisting",
            "component": "*",
            "framework": "*",
            "data": {
                "templateName": "defaultTemplate",
                "action": "homeListing",
                "fields": {
                    "demoframework1": {
                        "teacher": [
                            {
                                "name": "observations",
                                "icon": {
                                    "web": "assets/images/mask-image/observation_category.png",
                                    "app": "assets/imgs/ic_observation_category.png"
                                }
                            }
                        ]
                    },
                    "stateboard": {
                        "teacher": [
                            {
                                "name": "observations",
                                "icon": {
                                    "web": "assets/images/mask-image/observation_category.png",
                                    "app": "assets/imgs/ic_observation_category.png"
                                }
                            }
                        ]
                    },
                    "cbse": {
                        "teacher": [
                            {
                                "name": "observations",
                                "icon": {
                                    "web": "assets/images/mask-image/observation_category.png",
                                    "app": "assets/imgs/ic_observation_category.png"
                                }
                            }
                        ]
                    }
                }
            },
            "created_on": "2024-04-04T10:04:30.491Z",
            "last_modified_on": "2024-03-21T06:05:05.456Z",
            "rootOrgId": "0138786843639234561"
        }
    },
    "ts": "2024-04-11T09:00:39.116Z",
    "ver": "1.0"
}`

**Reply by 5Amogh** [reply]
@siddhishinde0723 Can you please update "observations" to "observation" in the formconfig and check?

**Reply by siddhishinde0723** [reply]
@5Amogh Thanks for the help; this is solved now.

**Comment by siddhishinde0723** [comment]
@5Amogh I am also trying to add a project in "Browse Other Categories." i have also updated the form but still the project is not visible.

Here is the form read 
`  "data": {
                "templateName": "defaultTemplate",
                "fields": {
                    "NCF": {
                        "teacher": [
                            {
                                "name": "observation",
                                "icon": {
                                    "web": "assets/images/mask-image/observation_category.png",
                                    "app": "assets/imgs/ic_observation_category.png"
                                }
                            },
                            {
                                "name": "project",
                                "icon": {
                                    "web": "",
                                    "app": "assets/imgs/ic_project.svg"
                                }
                            }
                        ]
                    }
                }
            }` 

Can you please help.

cc: @kiranharidas187

**Reply by 5Amogh** [reply]
@siddhishinde0723 Are you trying to check project on Portal? Project & Survey are not available on Portal.
Cc: @kiranharidas187

**Reply by siddhishinde0723** [reply]
@5Amogh I am trying to check on Mobile app.

**Reply by 5Amogh** [reply]
@vishwanath1004 can you please help here

**Reply by siddhishinde0723** [reply]
@5Amogh This is solved the form was not getting updated.

Thanks.
